packages/h5wasm/src/utils.ts

```diff
@@ -19,73 +20,66 @@ export function isHDF5(buffer: ArrayBuffer): boolean {
+export function convertMetadataToDType(metadata: Metadata): DType {
+  if (isIntegerMetadata(metadata)) {
+    const { signed, size: length, littleEndian } = metadata;
+
+    return {
+      class: signed ? DTypeClass.Integer : DTypeClass.Unsigned,
+      size: length * 8,
+      endianness: littleEndian ? Endianness.LE : Endianness.BE,
+    };
+  }
 
-export function isH5WasmExternalLink(
-  entity: H5WasmEntity
-): entity is H5WasmExternalLink {
-  return entity instanceof H5WasmExternalLink;
-}
+  if (isFloatMetadata(metadata)) {
+    const { size: length, littleEndian } = metadata;
 
-export function assertH5WasmDataset(
-  entity: H5WasmEntity
-): asserts entity is H5WasmDataset {
-  if (!isH5WasmDataset(entity)) {
-    throw new Error('Expected H5Wasm entity to be dataset');
+    return {
+      class: DTypeClass.Float,
+      size: length * 8,
+      endianness: littleEndian ? Endianness.LE : Endianness.BE,
+    };
   }
-}
 
-export function assertH5WasmEntityWithAttrs(
-  entity: H5WasmEntity
-): asserts entity is H5WasmGroup | H5WasmDataset {
-  if (!isH5WasmGroup(entity) && !isH5WasmDataset(entity)) {
-    throw new Error('Expected H5Wasm entity with attributes');
-  }
-}
+  if (isStringMetadata(metadata)) {
+    const { size: length, cset } = metadata;
 
-export function convertMetadataToDType(metadata: Metadata): DType {
-  const { signed, type, size: length, littleEndian } = metadata;
+    return {
+      class: DTypeClass.String,
+      length,
+      charSet: cset === 1 ? 'UTF-8' : 'ASCII',
+    };
+  }
 
-  // See H5T_class_t in https://github.com/usnistgov/h5wasm/blob/main/src/hdf5_util_helpers.d.ts
-  switch (type) {
-    case 0:
-      return {
-        class: signed ? DTypeClass.Integer : DTypeClass.Unsigned,
-        size: length * 8,
-        endianness: littleEndian ? Endianness.LE : Endianness.BE,
-      };
+  if (isCompoundMetadata(metadata)) {
+    const { compound_type } = metadata;
+    const { members, nmembers } = compound_type;
 
-    case 1:
-      return {
-        class: DTypeClass.Float,
-        size: length * 8,
-        endianness: littleEndian ? Endianness.LE : Endianness.BE,
-      };
+    if (nmembers === 2 && members[0].name === 'r' && members[1].name === 'i') {
+      const [realTypeMetadata, imagTypeMetadata] = members;
+      assertNumericMetadata(realTypeMetadata);
+      assertNumericMetadata(imagTypeMetadata);
 
-    case 3:
       return {
-        class: DTypeClass.String,
-        length,
-        charSet: metadata.cset === 1 ? 'UTF-8' : 'ASCII',
+        class: DTypeClass.Complex,
+        realType: convertMetadataToDType(realTypeMetadata),
+        imagType: convertMetadataToDType(imagTypeMetadata),
       };
+    }
 
-    default:
-      return {
-        class: DTypeClass.Unknown,
-      };
+    return {
+      class: DTypeClass.Compound,
+      fields: Object.fromEntries(
+        members.map((member) => [member.name, convertMetadataToDType(member)])
+      ),
```

Comment:
Following our line of work in HSDS, we try to be as specific as possible even if it is not complex and that we don't provide visualizations for such types.